askama_shared/src/generator.rs

```diff
@@ -804,26 +804,34 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         buf.writeln(";")
     }
 
-    fn is_shadowing_variable(&self, var: &Target<'a>) -> bool {
+    fn is_shadowing_variable(&self, var: &Target<'a>) -> Result<bool, CompileError> {
         match var {
             Target::Name(name) => {
                 let name = normalize_identifier(name);
                 match self.locals.get(&name) {
                     // declares a new variable
-                    None => false,
+                    None => Ok(false),
                     // an initialized variable gets shadowed
-                    Some(meta) if meta.initialized => true,
+                    Some(meta) if meta.initialized => Ok(true),
                     // initializes a variable that was introduced in a LetDecl before
-                    _ => false,
+                    _ => Ok(false),
                 }
             }
             Target::Tuple(_, targets) => targets
                 .iter()
-                .any(|target| self.is_shadowing_variable(target)),
+                .find_map(|target| match self.is_shadowing_variable(target) {
+                    Ok(false) => None,
+                    outcome => Some(outcome),
+                })
+                .unwrap_or(Ok(false)),
             Target::Struct(_, named_targets) => named_targets
                 .iter()
-                .any(|(_, target)| self.is_shadowing_variable(target)),
-            _ => panic!("Cannot have literals on the left-hand-side of an assignment."),
+                .find_map(|(_, target)| match self.is_shadowing_variable(target) {
+                    Ok(false) => None,
+                    outcome => Some(outcome),
+                })
+                .unwrap_or(Ok(false)),
+            _ => Err("Cannot have literals on the left-hand-side of an assignment.".into()),
```

Comment:
Probably rewrite this as "literals are not allowed on the left-hand side of an assignment" (no capital, no period).

Reply:
Changed.